src/email_service/send_email/lambda_function.py

```diff
@@ -222,13 +221,24 @@ def lambda_handler(event, context):
             )
 
             if response["Count"] == 0:
-                # Run ID does not exist, save all emails to DynamoDB with PENDING status
-                spreadsheet_info = file_service.get_file_info(
-                    sqs_message["spreadsheet_file_id"], access_token
-                )
-                spreadsheet_s3_object_key = spreadsheet_info["s3_object_key"]
-                sheet_data, _ = read_sheet_data_from_s3(spreadsheet_s3_object_key)
-                logger.info("Read sheet data from S3: %s", sheet_data)
+                if recipient_source == "SPREADSHEET":
+                    spreadsheet_info = file_service.get_file_info(
+                        sqs_message["spreadsheet_file_id"], access_token
+                    )
+                    spreadsheet_s3_object_key = spreadsheet_info["s3_object_key"]
+                    sheet_data, _ = read_sheet_data_from_s3(spreadsheet_s3_object_key)
+                    logger.info("Read sheet data from S3: %s", sheet_data)
+                else:  # DIRECT mode
```

Comment:
So when the request body doesnt contain "recipient_source", for example like this : 
```
{
    "subject": "【參與證明】AWS Educate 雲端大使感謝您的參與 - local-dev 05:31 （參與證明請參見附件）",
    "display_name": "AWS Educate 參與證明",
    "template_file_id": "0aeff5f3f7cb44708f3706906d5d4fff",
    "spreadsheet_file_id": "",
    "is_generate_certificate": false,
    "reply_to": "harryup2000@gmail.com",
    "sender_local_part": "event",
    "recipients": [
        {
            "email": "harryup2000@gmail.com",
            "template_variables": {
                "Name": "Harry 老闆"
            }
        }
    ],
    "attachment_file_ids": [
        "d9df7cd2c6fa4fadbf82df5e0dc2612f"
    ],
    "bcc": [
        "rich.liu627@gmail.com",
        "harryup2000@gmail.com"
    ],
    "cc": [
        "ptqwe20020413@gmail.com",
        "h830519h@yahoo.com.tw"
    ]
}
```
recipient_source is `None` in this case, which means that it enters the else situation. 
為了向前兼容（以前沒有recipient_source），再加上我們default是"SPREADSHEET"，我們應該讓None這個情形導到"SPREADSHEET"，因此建議在212行加上判斷式

Reply:
Hi @chungchihhan 

In this situation, If the API Request body does not contain the `recipient_source`, it will be set to `SPREADSHEET` to maintain backward compatibility.

Line 282 in `validate_input/lambda_function.py` already sets the `recipient_source` to `SPREADSHEET` as the default value if it is not present in the API Reqeust body.


https://github.com/aws-educate-tw/aws-educate-tpet-backend/blob/8ebdeb16c67413db69e232d2ae8a7837283fc5cc/src/email_service/validate_input/lambda_function.py#L282

Reply:
Got it, LGTM !